agent/consul/server_test.go

```diff
@@ -1207,7 +1190,11 @@ func TestServer_Expect_NonVoters(t *testing.T) {
 	}
 
 	t.Parallel()
-	dir1, s1 := testServerDCExpectNonVoter(t, "dc1", 2)
+	dir1, s1 := testServerWithConfig(t, func(c *Config) {
```

Comment:
Does the datacenter not need to be set here?

Reply:
Good question! We always set the datacenter to `dc1` in `testServerWithConfig`. We were setting it multiple times before, so I removed the extra sets when removing the helper.